### ISSUE TITLE
Default fatal error handler might be not ideal

### ISSUE THREAD

**Issue by anykeyh** [arcage/crystal-email#56]
Hello,

I'm having an application in production and it took me hours to figure out why out of nowhere the application was dying.

It appears that `crystal-email` default handler for fatal error was the cause:

```crystal
  DEFAULT_FATAL_ERROR_HANDLER = OnFatalErrorProc.new do |e|
    if backtrace = e.backtrace?
      backtrace.each do |line|
        STDERR << "  from " << line << '\n'
      end
    end
    exit(1)
  end
```

I think the default handler should just reraise the error, not kill the program. 

In `crystal-email`, host timeout is a fatal error, when it can occur at anytime.

What do you think?

**Comment by arcage** [comment]
Thank you for your comment.

Although there must have been some reasons or other on that, I cannot remember them at now.

I will try to consider it again.

Until I conclude that, could you please override it by using `EMail::Client::Config#on_fatal_error=` ?

**Comment by anykeyh** [comment]
Yeah that's what I did as current fix.

The main problem is what is considered as fatal: 
- If the SMTP server is down or timeout, this is considered as fatal error and the default behavior of your shard is to stop the program. 

This should not be the default :smile:. So maybe you should consider FATAL error like configuration errors or other unrecoverable error?

**Comment by arcage** [comment]
I released the v0.5.0 including change the behavior of the default `on_fatal_error` handler.

The new one doesn't kill the program, but re-rise same exception.

**Comment by arcage** [comment]
I will close this issue once.

Please reopen this if the problem is unsolved.
